### PR TITLE
4514: Ensure that admin users can not be logged out

### PR DESCRIPTION
**Pull request by cableman**
#### Link to issue

https://platform.dandigbib.org/issues/4514

#### Description

If you get `ding_adgangsplatformen` enabled but not disabled `ding_user_form` and the other way around you will end up with a site that non-provider users can't login.

#### Screenshot of the result

No screenshot

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

No comments

### CODE REVIEW

File: ding2.install [hunk 1]
```diff
@@ -106,6 +106,13 @@ function ding2_install() {
     module_disable(array('overlay'));
   }
 
+  // Enable ding_user_form to allow provider login. It's enabled here as
+  //new/local sites should use "user form" and updated production should via
+  // update hooks turn on adgangsplatformen.
+  if (module_exists('ding_user_form')) {
+    module_enable(array('ding_user_form'));
+  }
```
Comment (kasperg) [inline ding2.install]: Why add this when [we allow the user to select which modules to enable during install - including `ding_user_form`](https://github.com/ding2/ding2/blob/5a7a2f9de2a3feb816e8c453a0c0874ce1cccdb8/ding2.profile#L378-L388)?

File: ding2.install [hunk 2]
```diff
@@ -106,6 +106,13 @@ function ding2_install() {
     module_disable(array('overlay'));
   }
 
+  // Enable ding_user_form to allow provider login. It's enabled here as
+  //new/local sites should use "user form" and updated production should via
```
Comment (kasperg) [inline ding2.install]:
```suggestion
  // new/local sites should use "user form" and updated production should via
```

Note the Scrutinizer error.

File: ding2.install (hunk 1 above)
Comment (holt83) [reply ding2.install]: `module_exists` will only return TRUE if the module is already enabled.

Comment (cableman) [reply ding2.install]: Fixed

File: modules/ding_user_form/ding_user_form.install [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+/**
+ * @file
+ * Handle module installation.
+ */
+
+/**
+ * Implements hook_install().
```
[the remark this answers is not in the document]
Comment (cableman) [reply modules/ding_user_form/ding_user_form.install]: Fixed

File: ding2.install (hunk 2 above)
Comment (cableman) [reply ding2.install]: Fixed

**Comment by kasperg** [comment]
Closed in favour of #1533.

**Pull request closed without merging**